lib/src/cli.rs

```diff
@@ -281,17 +285,39 @@ async fn upgrade(opts: UpgradeOpts) -> Result<()> {
     let commit = booted_deployment.csum();
     let state = ostree_container::store::query_image_commit(repo, &commit)?;
     let digest = state.manifest_digest.as_str();
-    let fetched = pull(repo, &imgref, opts.quiet).await?;
 
-    if fetched.merge_commit.as_str() == commit.as_str() {
-        println!("Already queued: {digest}");
+    if opts.check {
+        // pull the image manifest without the layers
+        let config = Default::default();
+        let mut imp = ostree_container::store::ImageImporter::new(repo, &imgref, config).await?;
+        match imp.prepare().await? {
+            PrepareResult::AlreadyPresent(c) => {
+                println!(
+                    "No changes available for {}. Latest digest: {}",
+                    imgref, c.manifest_digest
+                );
+            }
+            PrepareResult::Ready(p) => {
+                println!(
+                    "New manifest available for {}. Digest {}",
+                    imgref, p.manifest_digest
+                );
+            }
+        }
         return Ok(());
-    }
+    } else {
```

Comment:
Since we have a `return` above I don't think we need to indent everything here, right?  (Though if you follow my above suggestion, the touch_if_changed handling would be outside the `else`)

Reply:
I updated the PR with a change in the return statement and moved the touch_if_changed handling outside of the else. Now it will carry on and honor the option. 
However that keeps the indentation.